provider/storage_benchmark.py

```diff
@@ -0,0 +1,322 @@
+       fio benchmark.
+"""
+
+import logging
+import os
+import time
+import re
+
+from platform import machine
+
+from virttest import utils_misc
+
+from avocado.core import exceptions
+
+GIT_DOWNLOAD = 'git'
+CURL_DOWNLOAD = 'curl'
+
+TAR_UNPACK = 'tar'
+
+RANDOM_STRING = utils_misc.generate_random_string(8)
+STAMP = ''.join((time.strftime('%Y-%m-%d-%H-%M-%S'), RANDOM_STRING))
+
+
+class StorageBenchmark(object):
+    """
+    Create a Benchmark class which provider common interface(method) for
+    using Benchmark tool to run test.
+
+    """
+    cmds = {'linux': {'_symlinks': 'ln -s -f %s %s',
+                      '_list_pid': 'pgrep -xl %s',
+                      '_kill_pid': 'killall %s',
+                      '_rm_file': 'rm -rf {}'},
+            'windows': {'_symlinks': 'mklink %s %s',
+                        '_list_pid': 'TASKLIST /FI "IMAGENAME eq %s',
+                        '_kill_pid': 'TASKKILL /F /IM %s /T',
+                        '_rm_file': 'RD /S /Q "{}"'}}
+    tar_map = {'.tar': '-xvf', '.tar.gz': '-xzf',
+               '.tar.bz2': '-xjf', '.tar.Z': '-xZf'}
+    download_cmds = {GIT_DOWNLOAD: 'rm -rf {0} && git clone {1} {0}',
+                     CURL_DOWNLOAD: 'curl -o {0} {1}'}
+    unpack_cmds = {TAR_UNPACK: '_tar_unpack_file'}
+
+    def __init__(self, os_type, session, name):
+        """
+        :param session: session connected guest
+        :param name: the name of benchmark
+        :type name: str
+        """
+        self.session = session
+        self.name = name
+        self.os_type = os_type
+        self.env_files = []
+
+    def __getattr__(self, item):
+        for k, v in self.cmds[self.os_type].items():
+            if k == item:
+                return v
+
+    def session_cmd(self, cmd, timeout=60):
+        """ Session command. """
+        self.session.cmd(cmd, timeout)
+
+    def make_symlinks(self, src, dst):
+        """
+        Make symlinks between source file and destination file by force.
+
+        :param src: source file
+        :type src: str
+        :param dst: destination file
+        :type dst: str
+        """
+        self.session.cmd(self._symlinks % (src, dst))
+        self.env_files.append(dst)
+
+    def _wait_procs_done(self, timeout=1800):
+        """
+        Wait all the processes are done.
+
+        :param timeout: timeout for waiting
+        :type timeout: float
+        """
+        logging.info('Checking the currently running %s processes.' % self.name)
+        if not utils_misc.wait_for(
+                lambda: self.name not in self.session.cmd_output(
+                    self._list_pid % self.name), timeout, step=3.0):
+            raise exceptions.TestError(
+                'All %s processes are not done in %s sec.' % (self.name, timeout))
+
+    def _kill_procs(self):
+        """Kill the specified processors by force."""
+        logging.info('Killing all %s processes by force.' % self.name)
```

Comment:
Should judge if processes existed at first, 
or the cmd execution would be failed if process not existed.
# killall -9 qemu-kvm
qemu-kvm: no process found
# echo $?
1

Reply:
```
>>> session.cmd_output('unamed -sr')
"bash: unamed: command not found...\nSimilar command is: 'uname'\n"
>>> session.cmd('unamed -sr')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/root/vmt/kar/workspace/lib/python3.6/site-packages/aexpect/client.py", line 1288, in cmd
    raise ShellCmdError(cmd, s, o)
aexpect.exceptions.ShellCmdError: Shell command failed: 'unamed -sr'    (status: 127,    output: "bash: unamed: command not found...\nSimilar command is: 'uname'\n")
>>> 

It's not raise errors.
```